lib/s3middleware/azureHelpers/mpuUtils.js

```diff
@@ -80,5 +87,115 @@ azureMpuUtils.getSubPartSize = (subPartInfo, subPartIndex) => {
         lastPartSize : azureMpuUtils.maxSubPartSize;
 };
 
+azureMpuUtils.getSubPartIds = (part, uploadId) =>
+    [...Array(part.numberSubParts).keys()].map(subPartIndex =>
+        azureMpuUtils.getBlockId(uploadId, part.partNumber, subPartIndex));
+
+azureMpuUtils.putSinglePart = (azureClient, request, params, dataStoreName,
+log, cb) => {
+    const { bucketName, partNumber, size, objectKey, contentMD5, uploadId }
+        = params;
+    const totalSubParts = 1;
+    const blockId = azureMpuUtils.getBlockId(uploadId, partNumber, 0);
+    const passThrough = new stream.PassThrough();
+    const options = {};
+    if (contentMD5) {
+        options.useTransactionalMD5 = true;
+        options.transactionalContentMD5 =
+            objectUtils.getBase64MD5(contentMD5);
+    }
+    request.pipe(passThrough);
+    return azureClient.createBlockFromStream(blockId,
+    bucketName, objectKey, passThrough, size, options,
+    (err, result) => {
+        if (err) {
+            log.error('Error from Azure data backend uploadPart',
+                { err: err.message, dataStoreName });
+            return cb(errors.InternalError.customizeDescription(
+                `Error returned from Azure: ${err.message}`)
+            );
+        }
+        const eTag = objectUtils.getHexMD5(result.headers['content-md5']);
+        return cb(null, eTag, totalSubParts);
+    });
+};
+
+azureMpuUtils.putNextSubPart = (azureClient, partParams, subPartInfo,
+subPartStream, subPartIndex, resultsCollector) => {
+    const { uploadId, partNumber, bucketName, objectKey } = partParams;
+    const subPartSize = azureMpuUtils.getSubPartSize(
+        subPartInfo, subPartIndex);
+    const subPartId = azureMpuUtils.getBlockId(uploadId, partNumber,
+        subPartIndex);
+    resultsCollector.pushOp();
+    azureClient.createBlockFromStream(subPartId, bucketName,
+    objectKey, subPartStream, subPartSize, {}, err => {
+        resultsCollector.pushResult(err, subPartIndex);
+    });
+};
+
+azureMpuUtils.putSubParts = (azureClient, request, params, dataStoreName,
+log, cb) => {
+    const subPartInfo = azureMpuUtils.getSubPartInfo(params.size);
+    const resultsCollector = new ResultsCollector();
+    const hashedStream = new MD5Sum();
+    const streamInterface = new SubStreamInterface(hashedStream);
+    log.trace('data length is greater than max subpart size;' +
+        'putting multiple parts');
+
+    resultsCollector.on('error', (err, subPartIndex) => {
+        streamInterface.stopStreaming(request);
+        log.info(`Error putting subpart to Azure: ${subPartIndex}`,
+            { err: err.message, dataStoreName });
+        return cb(errors.InternalError.customizeDescription(
+            `Error returned from Azure: ${err}`));
+    });
+
+    resultsCollector.on('done', (err, results) => {
+        if (err) {
+            log.info('Error putting last subpart to Azure',
```

Comment:
Same here, why not use error level. Also the error property needs to be set instead of err.